src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -50,46 +52,53 @@ private enum AttributeKind
         }
 
         private static readonly char[] s_illegalCharactersInSimpleName = { '/', '\\', ':' };
-        private string _input;
+        private ReadOnlySpan<char> _input;
         private int _index;
+        private string? _inputAsString;
```

Comment:
I am not sure we need `_inputAsString`. The only use seems to avoid `ToString` when throwing.

Reply:
Ok, deleted. I was not sure whether we really need it either.